depends/Makefile

```diff
@@ -133,9 +133,10 @@ endef
 
 define check_or_remove_sources
   mkdir -p $($(package)_source_dir); cd $($(package)_source_dir); \
-  $(build_SHA256SUM) -c $($(package)_fetched) >/dev/null 2>/dev/null || \
-    ( if test -f $($(package)_all_sources); then echo "Checksum missing or mismatched for $(package) source. Forcing re-download."; fi; \
-      rm -f $($(package)_all_sources) $($(1)_fetched))
+  ( $(build_SHA256SUM) --check --quiet --ignore-missing $($(package)_fetched) && \
```

Comment:
We use two different tools across the three builder platforms:
https://github.com/zcash/zcash/blob/35bb61be3900ff9d635de550c9efbe6b4ce96192/depends/builders/linux.mk#L1
https://github.com/zcash/zcash/blob/35bb61be3900ff9d635de550c9efbe6b4ce96192/depends/builders/freebsd.mk#L1
https://github.com/zcash/zcash/blob/35bb61be3900ff9d635de550c9efbe6b4ce96192/depends/builders/darwin.mk#L9

So any general flags we specify here must be supported by both `sha256sum` and `shasum`.

Reply:
Good catch. Somehow I missed those definitions when doing a grep -r, so I thought the only case was `sha256sum`. Hm, maybe I should revert to the original approach and just fix the `test -f` call with the `xargs` pipe fix. Thoughts?

Also, is it possible for me to conveniently test builds on all of those platforms with CI before PR review?

Reply:
Oh, another issue, that maybe deserves a cleanup pr: why are many expansions here using the `$($(package)_blah)` style while some are using `$($(1)_foo)` style? Are they correctly distinct?

My first read of this would suggest that when the multi-line `define` is expanded it is always with the first parameter being `$(package)` and therefore both styles lead to the same result. If that's true then I would want to do a cleanup to change them all to `$($(1)_foo)` style.

OTOH, if there's some subtle scoping reason for different reference styles, it'd be good to add a comment about the gotcha there.

If we're confident on the next step for that issue we could address them both in a single PR.

Reply:
A recent version of FreeBSD's `shasum` seems to support these options: https://www.freebsd.org/cgi/man.cgi?query=shasum&sektion=1&manpath=freebsd-release-ports

@str4d says that `shasum` 5.9.6 does not support `--quiet` or `--ignore-missing` on Ubuntu 18.04 LTS. (But this will not be used because it's on Linux.)